org.eclipse.jdt.launching/launching/org/eclipse/jdt/internal/launching/LaunchingPlugin.java

```diff
@@ -582,6 +589,66 @@ public void saving(ISaveContext context1) throws CoreException {
 		ResourcesPlugin.getWorkspace().addResourceChangeListener(this, IResourceChangeEvent.PRE_DELETE | IResourceChangeEvent.PRE_CLOSE);
 		DebugPlugin.getDefault().getLaunchManager().addLaunchListener(this);
 		DebugPlugin.getDefault().addDebugEventListener(this);
+		IEclipsePreferences instanceNode = InstanceScope.INSTANCE.getNode(getBundle().getSymbolicName());
+		IEclipsePreferences defaultNode = DefaultScope.INSTANCE.getNode(getBundle().getSymbolicName());
+		if (instanceNode.getBoolean(PREF_DETECT_VMS_AT_STARTUP, defaultNode.getBoolean(PREF_DETECT_VMS_AT_STARTUP, true))) {
+			new Job(LaunchingMessages.lookupInstalledJVMs) {
+				@Override
+				protected IStatus run(IProgressMonitor monitor) {
+					Collection<File> rootDirectories = new HashSet<>();
+					rootDirectories.add(new File("/usr/lib/jvm")); //$NON-NLS-1$
```

Comment:
I assume this will do nothing for Windows, how does the preference page address this (I didn't see if there is some switch which doesn't show the new preference UIs / didn't see anything in the message to indicate no Windows support)?

Reply:
This particular directory will be ignored. I think the `.sdkman` folder is also used on Windows; if so the job can detect such installations on Windows.
However, if you can suggest some default directories to look at for Windows, I'd be glad to add them to this list. I just happen to not know how a JDK in installed usually on Windows.